### ISSUE TITLE
OnPrem - "Schema Deployment From Data Files" works with SQL, but not SQL CE

### ISSUE THREAD

**Issue by md-hc** [umbraco/Umbraco.Deploy.Issues#67]
### Bug summary

Following the [documentation](https://our.umbraco.com/documentation/Add-ons/Umbraco-Deploy/Installing-Deploy/Existing-site/) for installing on an existing site, "Schema Deployment From Data Files" causes errors with SQL CE, but works OK with normal SQL.

### Specifics

Using OnPrem 4.1.3, schema was extracted from an existing DB. The schema works fine using a SQL DB or Azure DB, and we are able to sync content between environments.

However when we work locally using SQL CE, running "Schema Deployment From Data Files" results in the following error:

```
EXCEPTION
Umbraco.Deploy.Exceptions.ProcessArtifactException: Process pass #2 failed for artifact umb://document-type/34ced241a1554c6baa41b89c168be6d8. ---> System.Data.SqlServerCe.SqlCeException: A duplicate value cannot be inserted into a unique index. [ Table name = cmsPropertyTypeGroup,Constraint name = PK_cmsPropertyTypeGroup ]
```

This does not appear to be document type specific, as deleting this document type causes the same error with others.

It also does not appear to be related to any packages as the erroring document types do not contain any data types using those packages.

### Steps to reproduce

**Please note - these steps are project specific. The relevant schema files and required files to replicate have been sent to Alina at Umbraco Support.**

- Setup a clean Umbraco 8.15.1 project, with no starter kit and using SQL CE
- Install-Package UmbracoDeploy.OnPrem -Version 4.1.3
- Install-Package Our.Umbraco.GMaps -Version 1.3.3
- Install-Package UmbracoForms -Version 8.7.6
- Install-Package tooorangey.uEditorNotes -Version 8.0.1
- Make sure the "HtmlEditor" App_Plugin is added
- Run "Schema Deployment From Data Files", it will fail
- Repeat the same process but using a SQL DB (tested with 15.0.2000.5) instead of CE, import will run successfully

**Comment by AndyButland** [comment]
After a bit of digging, it looks like you've run into [this issue](https://github.com/umbraco/Umbraco-CMS/issues/10381), found in the version of the CMS you are using.  From what I read it's resolved for 8.16.

In the meantime there's a workaround shared in the issue, which I'll repeat here as I've tried it and was able to get the schema extraction to run to completion once I'd followed it:

- Install something to allow you to run queries on SQL CE, I used CompactView which you can download from [here](https://sourceforge.net/projects/compactview/).
- Make sure you also have SQL CE Compact V4 installed (if not you'll get an error message when you try to open the database using CompactView.  You can download that from [here](https://www.microsoft.com/en-us/download/details.aspx?id=30709).
- Open the `App_Data\Umbraco.sdf` file.
- Select to "Allow Editing"
- Run the following:

```
ALTER TABLE [cmsPropertyTypeGroup] ALTER COLUMN [Id] IDENTITY (56,1)
```

After that you should find an extraction works as expected.

**Comment by md-hc** [comment]
Thank you very much @AndyButland !
